src/EnergyPlus/DataLoopNode.hh

```diff
@@ -108,82 +101,829 @@ namespace DataLoopNode {
+                                                                                                                "ZoneExhaust",
+                                                                                                                "Setpoint",
+                                                                                                                "Electric",
+                                                                                                                "OutsideAirReference",
+                                                                                                                "InducedAir"};
 
-        case NodeFluidType::Electric:
-            return "Electric";
+    constexpr static std::array<std::string_view, static_cast<int>(ConnectionType::Num)> ConnectionTypeNamesUC = {"BLANK",
+                                                                                                                  "INLET",
+                                                                                                                  "OUTLET",
+                                                                                                                  "INTERNAL",
+                                                                                                                  "ZONENODE",
+                                                                                                                  "SENSOR",
+                                                                                                                  "ACTUATOR",
+                                                                                                                  "OUTDOORAIR",
+                                                                                                                  "RELIEFAIR",
+                                                                                                                  "ZONEINLET",
+                                                                                                                  "ZONERETURN",
+                                                                                                                  "ZONEEXHAUST",
+                                                                                                                  "SETPOINT",
+                                                                                                                  "ELECTRIC",
+                                                                                                                  "OUTSIDEAIRREFERENCE",
+                                                                                                                  "INDUCEDAIR"};
 
-        default:
-            return "blank";
-        }
-    }
-
-    constexpr const char *ValidConnectionTypes(NodeConnectionType const NodeConnectionType) // Valid Connection Types for Nodes
+    enum class ConnectionObjectType
```

Comment:
New enum with values mined out of the BND files. I expect that this is not yet comprehensive.

Reply:
> New enum with values mined out of the BND files. I expect that this is not yet comprehensive.

@mitchute Is this supposed to be a subset of all object types that only includes objects that might register a node connection?

Reply:
I suppose it's just the objects that register node connections at this point. Good clarification.